cli/command/load_balancer_test.go

```diff
@@ -0,0 +1,491 @@
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": getCLIContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := loadBalancerCommand.delete(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestDropPortFromLoadBalancer(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	ports := []models.Port{
+		models.Port{
+			ContainerPort: 80,
+			HostPort:      80,
+			Protocol:      "tcp",
+		},
+		models.Port{
+			ContainerPort: 81,
+			HostPort:      81,
+			Protocol:      "tcp",
+		},
+	}
+	healthCheck := models.HealthCheck{}
+	loadBalancer := &models.LoadBalancer{
+		HealthCheck:      healthCheck,
+		LoadBalancerID:   "lid",
+		LoadBalancerName: "lname",
+		Ports:            ports,
+	}
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(loadBalancer, nil)
+
+	ports = ports[:1]
+
+	req := models.UpdateLoadBalancerRequest{
+		HealthCheck:    nil,
+		LoadBalancerID: "lid",
+		Ports:          &ports,
+	}
+
+	base.Client.EXPECT().
+		UpdateLoadBalancer(req).
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "lid",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(&models.LoadBalancer{}, nil)
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	c := getCLIContext(t, args{"lname", "81"}, flags{})
+	if err := loadBalancerCommand.dropport(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDropPortFromLoadBalancer_UserInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": getCLIContext(t, nil, nil),
+		"Missing PORT arg": getCLIContext(t, args{"name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := loadBalancerCommand.dropport(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+
+}
+
+func TestDisplayHealthCheck(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	loadBalancer := &models.LoadBalancer{
+		LoadBalancerID:   "lid",
+		LoadBalancerName: "lname",
+	}
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(loadBalancer, nil)
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	c := getCLIContext(t, args{"lname"}, flags{})
+	if err := loadBalancerCommand.healthcheck(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUpdateHealthCheck(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	ports := []models.Port{}
```

Comment:
ditto: we don't care about ports in this test. The field should be `nil` and ignored for this test

Reply:
:white_check_mark: